Modules/Core/Common/include/itkVectorContainer.h

```diff
@@ -44,10 +48,10 @@ namespace itk
  * \sphinxexample{Core/Common/IterateOnAVectorContainer,Iterate On A Vector Container}
  * \endsphinx
  */
-template <typename TElementIdentifier, typename TElement>
+template <typename T1, typename T2 = void>
 class ITK_TEMPLATE_EXPORT VectorContainer
   : public Object
-  , private std::vector<TElement>
+  , private std::vector<std::conditional_t<std::is_void_v<T2>, T1, T2>>
```

Comment:
Unfortunately this "trick" does not work as smoothly as I was hoping for.... The compiler still considers `VectorContainer<ElementType>` and `VectorContainer<SizeValueType, ElementType>` to be different types. Which disallows interchanging one with the other. (For example, assigning a pointer to another.) Not really convenient 🤔

Reply:
Resolved by making `itk::VectorContainer` an _alias_ _template_ of `itk::ClassTemplate::VectorContainer`!